splice/queries/distribution.py

```diff
@@ -154,3 +164,48 @@ def get_possible_distributions(today=None, channel_id=None):
             "force_upload": True
         })
     return artifacts
+
+
+_SENTINEL = object()
+
+
+def multiplex_directory_tiles(tiles):
+    """Directory tile multiplexer that creates all possible combinations of the
+    tile sets based on its type and the target url. Given multiple sponsored tiles,
+    it'll pick one of them and create a new tile set with other directory tiles.
+    It also multiplexes the tiles with the same TLD+1 target url (e.g. www.mozilla.org).
+    Therefore, given a tile set with N sponsored tiles and M identical TLD+1 tiles, the
+    total number of multiplexed directory tile set is N*M.
+
+    Note: this function will NOT handle the same tld+1 URL between the sponsored and
+    non-sponsered tiles.
+
+    Params:
+        tiles: a list of tiles.
+    Return:
+        A list of tile sets. Note that the sponsored directory tile is always the
+        3rd entry in the list, other tiles will be sorted by the tile id.
+    """
+    sponsored = []
+    tlds = defaultdict(list)
+    for tile in tiles:
+        if tile["type"] == "sponsored":
+            sponsored.append(tile)
+        else:
+            url = furl(tile["url"]).netloc
+            assert url, "URL: %s is invalid" % tile["url"]
+            tlds[url].append(tile)
```

Comment:
Actually, it looks like you are using the full domain name... So I am not sure what you meant by TLD+1. I thought it meant the TLD (for example, `.com`) + one subdomain level (for example, `mozilla`).

But, you are doing this:

```
In [4]: furl('http://a.bunch.of.subdomains.support.mozilla.org').netloc
Out[4]: 'a.bunch.of.subdomains.support.mozilla.org'
```